nixos/modules/services/mail/listmonk/default.nix

```diff
@@ -0,0 +1,181 @@
+    (concatStringsSep "\n" (mapAttrsToList setDatabaseOption (if (cfg.databaseSettings != null) then cfg.databaseSettings else {})));
+  updateDatabaseConfigScript = pkgs.writeShellScriptBin "update-database-config.sh" ''
+    ${if cfg.mutableDatabaseSettings then ''
+      if [ ! -f $STATE_DIRECTORY/.db_settings_initialized ]; then
+        ${pkgs.postgresql}/bin/psql -d listmonk -f ${updateDatabaseConfigSQL} ;
+        touch $STATE_DIRECTORY/.db_settings_initialized
+      fi
+    '' else "${pkgs.postgresql}/bin/psql -d listmonk -f ${updateDatabaseConfigSQL}"
+    }
+  '';
+in
+{
+  ###### interface
+  options = {
+    services.listmonk = {
+      enable = mkEnableOption "Listmonk";
+      database.createLocally = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc "Create the PostgreSQL database and database user locally.";
+      };
+      package = mkOption {
+        type = types.package;
+        default = pkgs.listmonk;
+        defaultText = lib.literalExpression "pkgs.listmonk";
+        description = lib.mdDoc "Package to use for this service";
+      };
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = tomlFormat.type;
+        };
+        description = lib.mdDoc ''Static settings set in the config.toml, see <https://github.com/knadh/listmonk/blob/master/config.toml.sample> for details.
+        You can set secrets using the secretFile option with environment variables following <https://listmonk.app/docs/configuration/#environment-variables>.'';
+      };
+      mutableDatabaseSettings = mkOption {
+        type = types.bool;
+        default = true;
+        description = lib.mdDoc ''Database settings will be reset to the value set in this module if this is not enabled.
+        Enable this if you want to persist changes you have done in the application.
+        '';
+      };
+      databaseSettings = mkOption {
+        default = null;
+        type = with types; nullOr (submodule {
+          freeformType = oneOf [ (listOf str) (listOf (attrsOf anything)) str int bool ];
+
+          options = {
+            "app.notify_emails" = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "Administrator emails for system notifications";
+            };
+
+            "privacy.exportable" = mkOption {
+              type = listOf str;
+              default = ["profile" "subscriptions" "campaign_views" "link_clicks"];
+              description = lib.mdDoc "List of fields which can be exported through an automatic export request";
+            };
+
+            "privacy.domain_blocklist" = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "E-mail addresses with these domains are disallowed from subscribing.";
+            };
+
+            smtp = mkOption {
+              type = listOf (submodule (import ./smtp.nix));
```

Comment:
Normally we inline this and since this module is quite small I think we should do that here, too.

Reply:
I planned to reuse it for `bounce-mailboxes` though, but yeah, we can always do it later.

Reply:
Done.